src/System.Device.Gpio/System/Device/I2c/I2cDevice.cs

```diff
@@ -35,7 +35,12 @@ public static I2cDevice Create(I2cConnectionSettings settings)
     /// Reads a byte from the I2C device.
     /// </summary>
     /// <returns>A byte read from the I2C device.</returns>
-    public abstract byte ReadByte();
+    public virtual unsafe byte ReadByte()
+    {
+        Span<byte> toRead = stackalloc byte[1];
+        Read(toRead);
+        return toRead[0];
```

Comment:
```suggestion
        byte value = 0;
        Span<byte> toRead = new Span<byte>(&value, 1);
        Read(toRead);
        return value;
```
Is a bit cheaper (no stack-cookie, less zeroing of memory).
And `unsafe` can be kept 😉.

Reply:
I agree, this is a possible alternative. Choose either.